arangod/Aql/Functions.cpp

```diff
@@ -6244,6 +6244,14 @@ AqlValue Functions::GeoPolygon(ExpressionContext* expressionContext,
   builder->close();  // coordinates
   builder->close();  // object
 
+  // Now actually parse the result with S2:
+  geo::ShapeContainer container;
```

Comment:
Perhaps this part down until line 6254, as is repeated  in line 6330 for the multipolygon constructor, could be put in a lambda or something for being reused?

Reply:
Actually, it is `parseMultiPolygon` there, so it is genuinely different. And the two are not in the same function, so a lambda is not possible. I think I prefer it the way it is.